app/src/main/java/org/covidwatch/android/ui/exposures/ExposuresViewModel.kt

```diff
@@ -36,17 +37,14 @@ class ExposuresViewModel(
         }
     }
 
-    fun toggleExposureNotifications() {
+    fun enableExposureNotification(enable: Boolean) {
         viewModelScope.launch {
-            // Check if we need to stop the service
-            if (enManager.isEnabled().result() == true) {
-                enManager.stop()
-            } else { // Otherwise run the service
-                enManager.start().result()
-            }
+            val isEnabled = enManager.isEnabled().result() ?: false
 
-            // Handle the case of is service running or not simply by asking the manager
-            _exposureNotificationEnabled.value = enManager.isEnabled().result()
+            when {
+                enable && !isEnabled -> enManager.start().result()
```

Comment:
This is very elegant

Reply:
Thank you :)
Although, to me it looks a bit complex. I had troubles with `Switch` 2-way data binding so I used less cool approach.